HotFix/Transport/TcpTransport.cs

```diff
@@ -33,17 +34,32 @@ public static TcpTransport Initiate(IClock clock, string address, int port)
         public static TcpTransport Accept(IClock clock, string address, int port)
         {
             var endpoint = new IPEndPoint(IPAddress.Parse(address), port);
-            var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp) { ReceiveTimeout = 1000, NoDelay = true };
-
-            socket.EnableFastPath();
-            socket.Bind(endpoint);
-            socket.Listen(1);
-
-            var client = socket.Accept();
-
-            socket.Dispose();
-
-            return new TcpTransport(clock, client);
+            using (var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp) { ReceiveTimeout = 1000, NoDelay = true })
```

Comment:
I'm not sure if I didn't understand the documentation correctly, but `EndAccept` is meant to be called in the callback you provide to `BeginAccept` and it blocks until a connection is established. At that point the method returns with the connected `Socket`.

If we want to cancel the operation we need to close the original socket, using `Socket.Close` (which is pretty much a wrapper for `Dispose`).